service/src/main/java/com/fecklessweasel/service/objectmodel/Course.java

```diff
@@ -55,6 +58,33 @@ public static Course create(Connection conn, Department department, int courseNu
     }
 
     /**
+     *@author Hayden Schmackpfeffer
+     *@param conn SQL Connection
+     *@param dept Department which all returned courses belong to
+     *@return list of all courses in dept
+     */
+    public static ArrayList<Course> getAllWithDept(Connection conn, Department dept) throws ServiceException {
```

Comment:
Why return array list when you can return Iterable<Course> or simply generic List<Course>. The caller doesn't care the implementation of our List and we are making for harder future refactors.